Core/include/Acts/Geometry/ConeVolumeBounds.hpp

```diff
@@ -0,0 +1,228 @@
+  /// @param innerOffsetZ The tip  z position in of the inner cone, w.r.t center
+  /// @param outerAlpha  The opening angle of the outer cone (0 if no cone)
+  /// @param outerOffsetZ The tip  z position in of the outer cone, w.r.t center
+  /// @param halflengthZ The minimum z value of the inner and outer cones
+  /// @param avergePhi The phi orientation of the sector
+  /// @param halfPhiSector The opening angle phi sector
+  ConeVolumeBounds(double innerAlpha, double innerTipZ, double outerAlpha,
+                   double outerOffsetZ, double halflengthZ, double averagePhi,
+                   double halfPhiSector) noexcept(false);
+
+  /// Constructor - for general cylidner-cone setups
+  ///
+  /// @param cylinderR The inner radius of the cylinder
+  /// @param alpha  The opening angle of the cone (0 if no cone)
+  /// @param offsetZ The tip  z position in of the cone, w.r.t center
+  /// @param halflengthZ The minimum z value of the inner and outer cones
+  /// @param avergePhi The phi orientation of the sector (defaulted to 0)
+  /// @param halfPhiSector The opening angle phi sector
+  ///
+  /// @note depending on cylinderR > coneR it is constructing a cone with
+  /// cylindrical cutout or a cylinder with conical cutout
+  ConeVolumeBounds(double cylinderR, double alpha, double offsetZ,
+                   double halflengthZ, double averagePhi,
+                   double halfPhiSector) noexcept(false);
+
+  /// Constructor - from a fixed size array
+  ///
+  /// @param values The bound values
+  ConeVolumeBounds(const std::array<double, eSize>& values) noexcept(false)
+      : m_values(values) {
+    checkConsistency();
+    buildSurfaceBounds();
+  }
+
+  ConeVolumeBounds(const ConeVolumeBounds& cobo) = default;
+
+  ~ConeVolumeBounds() override = default;
+
+  ConeVolumeBounds& operator=(const ConeVolumeBounds& cobo) = default;
+
+  VolumeBounds::BoundsType type() const final { return VolumeBounds::eCone; }
+
+  /// Return the bound values as dynamically sized vector
+  ///
+  /// @return this returns a copy of the internal values
+  std::vector<double> values() const final;
+
+  /// This method checks if position in the 3D volume
+  /// frame is inside the cylinder
+  ///
+  /// @param pos is the position in volume frame to be checked
+  /// @param tol is the absolute tolerance to be applied
+  bool inside(const Vector3D& pos, double tol = 0.) const final;
+
+  /// Oriented surfaces, i.e. the decomposed boundary surfaces and the
+  /// according navigation direction into the volume given the normal
+  /// vector on the surface
+  ///
+  /// @param transform is the 3D transform to be applied to the boundary
+  /// surfaces to position them in 3D space
+  ///
+  /// It will throw an exception if the orientation prescription is not adequate
+  ///
+  /// @return a vector of surfaces bounding this volume
+  OrientedSurfaces orientedSurfaces(
+      const Transform3D* transform = nullptr) const override;
```

Comment:
This should also be `final`, right?

Reply:
Unless you had a extended class of ConeVolumeBounds that would change that.

Reply:
Changed.